spec/lexers/kotlin_spec.rb

```diff
@@ -22,5 +22,134 @@
+          ['Punctuation', '()']
+      end
+
+      it 'recognizes generic function' do
+        assert_tokens_equal 'fun <T, ClassT> makeField()',
+          ['Keyword', 'fun'],
+          ['Text', " "],
+          ['Punctuation', '<'],
+          ['Name.Class', 'T'],
+          ['Punctuation', ','],
+          ['Text', " "],
+          ['Name.Class', 'ClassT'],
+          ['Punctuation', '>'],
+          ['Text', " "],
+          ['Name.Function', 'makeField'],
+          ['Punctuation', '()']
+      end
+
+      it 'recognizes generic extension function' do
+        assert_tokens_equal 'fun <T, ClassT> String.makeField()',
+          ['Keyword', 'fun'],
+          ['Text', " "],
+          ['Punctuation', '<'],
+          ['Name.Class', 'T'],
+          ['Punctuation', ','],
+          ['Text', " "],
+          ['Name.Class', 'ClassT'],
+          ['Punctuation', '>'],
+          ['Text', " "],
+          ['Name.Class', 'String'],
+          ['Punctuation', '.'],
+          ['Name.Function', 'makeField'],
+          ['Punctuation', '()']
+      end
+
+      it 'recognizes function parameters' do
+        assert_tokens_equal 'fun makeField(thing: Int, other: String)',
+          ['Keyword', 'fun'],
+          ['Text', " "],
+          ['Name.Function', 'makeField'],
+          ['Punctuation', '('],
+          ['Name.Variable', 'thing'],
+          ['Punctuation', ':'],
+          ['Text', " "],
+          ['Name.Class', 'Int'],
+          ['Punctuation', ','],
+          ['Text', " "],
+          ['Name.Variable', 'other'],
+          ['Punctuation', ':'],
+          ['Text', " "],
+          ['Name.Class', 'String'],
+          ['Punctuation', ')']
+      end
+
+      it 'recognizes function with a generic parameter' do
+        assert_tokens_equal 'fun makeField(thing: Array<String>)',
+          ['Keyword', 'fun'],
+          ['Text', " "],
+          ['Name.Function', 'makeField'],
+          ['Punctuation', '('],
+          ['Name.Variable', 'thing'],
+          ['Punctuation', ':'],
+          ['Text', " "],
+          ['Name.Class', 'Array'],
+          ['Punctuation', '<'],
+          ['Name.Class', "String"],
+          ['Punctuation', '>)']
+      end
+
+      it 'recognizes function return type' do
+        assert_tokens_equal 'fun makeField(): String',
+          ['Keyword', 'fun'],
+          ['Text', " "],
+          ['Name.Function', 'makeField'],
+          ['Punctuation', '():'],
+          ['Text', " "],
+          ['Name.Class', 'String']
+      end
+
+      it 'recognizes function return generic type' do
+        assert_tokens_equal 'fun makeField(): Array<String>',
+          ['Keyword', 'fun'],
+          ['Text', " "],
+          ['Name.Function', 'makeField'],
+          ['Punctuation', '():'],
+          ['Text', " "],
+          ['Name.Class', 'Array'],
+          ['Punctuation', '<'],
+          ['Name.Class', 'String'],
+          ['Punctuation', '>']
+      end
+
+    end
```

Comment:
Same as #1001. You can remove these and make sure all examples are present in `samples/`

Reply:
Tests removed, samples were already present.